csrc/preseg_passes/move_pad.cpp

```diff
@@ -398,93 +399,93 @@ void propagatePads(Fusion* fusion) {
-    if (p->out()->uses().empty() && !p->out()->isFusionOutput()) {
+    if (pad->out()->uses().empty() && !pad->out()->isFusionOutput()) {
       continue;
     }
 
     // unify all consumer pad of tv;
-    auto* tv = p->in()->as<TensorView>();
-    for (Expr* use : tv->uses()) {
-      if (use == p) {
+    auto* pad_inp = pad->in()->as<TensorView>();
+    for (Expr* uses_of_pad_op_inp : pad_inp->uses()) {
+      if (uses_of_pad_op_inp == pad) {
         continue;
       }
       // check if use is the same pad operation (same pad value / width e.t.c.)
-      if (isSamePadOp(use, p)) {
+      if (isSamePadOp(uses_of_pad_op_inp, pad)) {
         // replace consumer of use->out() with p->out()
         ir_utils::replaceValInAllExprInputsAndFusionOutputs(
-            use->output(0), p->out());
+            uses_of_pad_op_inp->output(0), pad->out());
         // we could remove `use`, but `use` could still be in stack and needs to
         // be visited later. So push it to a vector and we'll remove it later.
-        pad_to_be_removed.push_back(use);
+        pad_to_be_removed.push_back(uses_of_pad_op_inp);
       }
     }
 
     // if tv is fusion output, we need to keep tv alive, it might render
     // propagating PadOp before tv->definition() being non-optimal.
-    if (tv->isFusionOutput()) {
+    if (pad_inp->isFusionOutput()) {
       continue;
     }
 
     // check for pad_dependencies to verify that 'p' can be moved before 'def'.
     std::unordered_set<Val*> pad_inputs;
-    for (Val* val : p->inputs()) {
-      if (val == p->in() || val->isConst()) {
+    for (Val* pad_inp : pad->inputs()) {
+      if (pad_inp == pad->in() || pad_inp->isConst()) {
         continue;
       }
-      pad_inputs.insert(val);
+      pad_inputs.insert(pad_inp);
     }
     std::unordered_set<Val*> pad_dependencies =
         DependencyCheck::getAllDependentVals(pad_inputs);
 
-    Expr* def = p->in()->definition();
+    Expr* def_of_pad_in = pad->in()->definition();
     Val* new_out = nullptr;
 
-    if (auto* uop = dynamic_cast<UnaryOp*>(def)) {
+    if (auto* uop = dynamic_cast<UnaryOp*>(def_of_pad_in)) {
       // check if unary op type is compatible for zero pad propagation.
       if (!zeroIsFixedPoint(uop->getUnaryOpType())) {
         continue;
       }
-      std::vector<Val*> new_pad_inputs = maybeMovePadBeforeDefinition(
-          tv, pad_dependencies, std::ref(stack), std::ref(simple_pad_set));
+      std::vector<Val*> outputs_of_moved_pad = maybeMovePadBeforeDefinition(
+          pad_inp, pad_dependencies, std::ref(stack), std::ref(simple_pad_set));
       // stop when move pad fails.
-      if (new_pad_inputs.empty()) {
+      if (outputs_of_moved_pad.empty()) {
         continue;
       }
       // update new outputs.
-      new_out =
-          ops::newValLike(new_pad_inputs[0], uop->out()->getDataType().value());
+      new_out = ops::newValLike(
+          outputs_of_moved_pad[0], uop->out()->getDataType().value());
```

Comment:
This is the only non-mechanical change and fixes the issue by making sure not to alter the same symbols for the extents.

Reply:
I don't see what's changed in this line.

Reply:
I guess unary op doesn't have that issue here, since we have only a single input.

Reply:
Sorry, you're right